moped-editor/src/views/projects/projectsMapView/components/MapDrawer.js

```diff
@@ -0,0 +1,180 @@
+import { drawerWidth } from "../../projectView/ProjectComponents";
+
+const openedMixin = (theme) => ({
+  transition: theme.transitions.create("width", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.enteringScreen,
+  }),
+  width: drawerWidth,
+  overflowX: "hidden",
+});
+
+const closedMixin = (theme) => ({
+  transition: theme.transitions.create("width", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  overflowX: "hidden",
+  width: theme.spacing(7),
+});
+
+/* Contain drawer within Paper wrapper (instead of full screen drawer) */
+const containDrawerMixin = () => ({
+  "& .MuiDrawer-root": {
+    position: "relative",
+  },
+  "& .MuiPaper-root": {
+    position: "relative",
+    borderTopLeftRadius: "4px",
+    borderBottomLeftRadius: "4px",
+  },
+});
+
+const Drawer = styled(MuiDrawer, {
+  shouldForwardProp: (prop) => prop !== "open",
+})(({ theme, open }) => ({
+  width: drawerWidth,
+  flexShrink: 0,
+  ...containDrawerMixin(),
+  ...(open && {
+    ...openedMixin(theme),
+    "& .MuiDrawer-paper": openedMixin(theme),
+  }),
+  ...(!open && {
+    ...containDrawerMixin(),
+    ...closedMixin(theme),
+    "& .MuiDrawer-paper": closedMixin(theme),
+  }),
+}));
+
+function DrawerContent({
+  children,
+  toggleDrawer,
+  open,
+  title,
+  showDrawerContent,
+}) {
+  const theme = useTheme();
+
+  return (
+    <>
+      <Grid container padding={theme.spacing(1)}>
+        <Grid container alignItems="center">
+          <Grid
+            item
+            flexGrow={1}
+            display={showDrawerContent && open ? "flex" : "none"}
+          >
+            <Typography variant="h2" color={theme.palette.text.primary}>
+              {title}
+            </Typography>
+          </Grid>
+          <Grid item>
+            <IconButton onClick={toggleDrawer}>
+              {open ? <ChevronLeftIcon /> : <ChevronRightIcon />}
+            </IconButton>
+          </Grid>
+        </Grid>
+      </Grid>
+      <Box
+        flexGrow={1}
+        display={showDrawerContent && open ? "flex" : "none"}
+        overflow="scroll"
+        padding={theme.spacing(1)}
+      >
+        {children}
+      </Box>
+    </>
+  );
+}
+
+/**
+ * Collapsible drawer for map filters.
+ * See https://mui.com/material-ui/react-drawer/#responsive-drawer
+ * See https://mui.com/material-ui/react-drawer/#mini-variant-drawer
+ */
+export default React.forwardRef(function MapDrawer({ children, title }, ref) {
+  /* Control desktop drawer and content visibility */
+  const [open, setOpen] = React.useState(true);
+  const [showDrawerContent, setShowDrawerContent] = React.useState(true);
+
+  const toggleDrawer = () => {
+    setOpen(!open);
+  };
+
+  const handleTransitionEnd = () => {
+    if (open) {
+      setShowDrawerContent(true);
+    } else {
+      // Repaint canvas on drawer close to avoid blank space in map
+      ref.current && ref.current.resize();
```

Comment:
The map and the map drawer are in a flex container. The map container grows to fill the view as the drawer shrinks when it collapses. The canvas within the map container that displays the map tile layer does not so we need to trigger the map to resize when the side drawer collapse transition is over. You can see the issue by commenting out this line when running locally.

Still not happy with the flicker/shift when this fires (you can see it on the right side of the map when toggling in desktop mode) but I think is fine for now. I'll keep it in mind as this grows, and I'm open to any suggestions. 🙏